src/gen/conditions.rs

```diff
@@ -2670,7 +2683,7 @@ fn test_spend_amount_exceeds_max() {
         cond_test("((({h1} ({h2} (0x010000000000000000 ())))")
             .unwrap_err()
             .1,
-        ErrorCode::InvalidCoinAmount
+        ErrorCode::AmountExceedsMaximum
```

Comment:
I'm thinking that it might be nice to preserve the error code for this case. This is the spend being invalid, because it's referring to a coin that cannot exist. We still convert this to `INVALID_CONDITION` when going back to python, but preserving this leaves the door open to record more precise errors.

I believe it would require mapping the error code on line 400.